core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -501,6 +501,10 @@ public TableMetadata replaceCurrentSnapshot(Snapshot snapshot) {
     return new Builder(this).setCurrentSnapshot(snapshot).build();
   }
 
+  public TableMetadata replaceCurrentSnapshot(long snapshotId) {
+    return new Builder(this).setCurrentSnapshot(snapshotId).build();
+  }
+
```

Comment:
I think this was an oversight, right? We don't need this new method; it is not used.
In fact, with the above change to `SnapshotProducer`, even `replaceCurrentSnapshot(Snapshot)` is no longer used and can be removed, I think.

Reply:
Yes, that's right. I missed this.